Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs

```diff
@@ -131,11 +131,54 @@ private string ConvertJToken(JToken jToken)
                         string => $"\"{value}\"",
                         _ => value?.ToString()?.Replace(',', '.'), // Replace decimal commas.
                     };
+
                 case JArray jArray:
-                    return $"new[] {{ {string.Join(", ", jArray.Select(ConvertJToken))} }}";
+                    var format = $"{string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth + 4))}";
+                    
+                    var items = jArray.Select(item => ConvertJToken(item, indentationDepth + 8)).ToList();
+                    
+                    // If the items are formatted (for ListValueOption) the don't inject line-by-line formatting
+                    if(items.Any(item => item.Contains(Environment.NewLine, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        var token = string.Join("", items);
+                        return $"new[]\n{format}{{\n{token}{format}}}";
+                    }
+                    // Otherwise, make sure that we have proper formatting for string arrays
+                    else
+                    {
+                        var stringArrayCodeBuilder = new StringBuilder("new[]");
+                        stringArrayCodeBuilder.AppendLine();
+                        stringArrayCodeBuilder.AppendLine($"{format}{{");
+
+                        var itemFormat = $"{string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth + 8))}";
+
+                        foreach (var item in items)
+                        {
+                            stringArrayCodeBuilder.AppendLine($"{itemFormat}{item},");
+                        }
+                        stringArrayCodeBuilder.Append($"{format}}}");
+
+                        return stringArrayCodeBuilder.ToString();
+                    }
+
                 case JObject jObject:
+                    var braceIndentation = string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth));
+                    var propertyIndentation = string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth + 4));
+                    if (jObject["name"] != null && jObject["value"] != null)
+                    {
+                        var objectCodeBuilder = new StringBuilder();
+                        objectCodeBuilder.AppendLine($"{braceIndentation}new ListValueOption");
+                        objectCodeBuilder.AppendLine($"{braceIndentation}{{");
+                        objectCodeBuilder.AppendLine($"{propertyIndentation}Name = \"{jObject["name"]}\",");
+                        objectCodeBuilder.AppendLine($"{propertyIndentation}Value = \"{jObject["value"]}\",");
+                        objectCodeBuilder.AppendLine($"{braceIndentation}}}");

```

Comment:
There's no comma between two `ListValueOptions`, see the below code generated from the Blog Post type with a two-item Text Field:

```c#
_contentDefinitionManager.AlterTypeDefinition("BlogPost", type => type
    .DisplayedAs("Blog Post")
    .Draftable()
    .Versionable()
    .WithPart("TitlePart", part => part
        .WithPosition("0")
    )
    .WithPart("AutoroutePart", part => part
        .WithPosition("1")
        .WithSettings(new AutoroutePartSettings
        {
            AllowCustomPath = true,
            Pattern = "{{ Model.ContentItem | container | display_text | slugify }}/{{ Model.ContentItem | display_text | slugify }}",
            ShowHomepageOption = false,
        })
    )
    .WithPart("BlogPost", part => part
        .WithPosition("3")
    )
    .WithPart("MarkdownBodyPart", part => part
        .WithPosition("2")
        .WithEditor("Wysiwyg")
    )
);

_contentDefinitionManager.AlterPartDefinition("BlogPost", part => part
    .WithField("Subtitle", field => field
        .OfType("TextField")
        .WithDisplayName("Subtitle")
        .WithPosition("0")
    )
    .WithField("Image", field => field
        .OfType("MediaField")
        .WithDisplayName("Banner Image")
        .WithPosition("1")
        .WithSettings(new ContentIndexSettings
        {
            Included = false,
            Stored = false,
            Analyzed = false,
            Sanitized = false,
            Tokenized = false,
            Template = "",
        })
        .WithSettings(new MediaFieldSettings
        {
            Multiple = false,
            AllowAnchors = true,
        })
    )
    .WithField("Tags", field => field
        .OfType("TaxonomyField")
        .WithDisplayName("Tags")
        .WithEditor("Tags")
        .WithDisplayMode("Tags")
        .WithPosition("2")
        .WithSettings(new TaxonomyFieldSettings
        {
            TaxonomyContentItemId = "4pjqtjhyxs1p965zhfhw3asqmy",
        })
    )
    .WithField("Category", field => field
        .OfType("TaxonomyField")
        .WithDisplayName("Category")
        .WithPosition("3")
        .WithSettings(new TaxonomyFieldSettings
        {
            TaxonomyContentItemId = "44z8pxhnwrvsgxzwzy96349cm5",
            Unique = true,
            LeavesOnly = true,
        })
    )
    .WithField("TextFieldTest", field => field
        .OfType("TextField")
        .WithDisplayName("TextFieldTest")
        .WithEditor("PredefinedList")
        .WithSettings(new TextFieldPredefinedListEditorSettings
        {
            Options = new[]
            {
                new ListValueOption
                {
                    Name = "Option1",
                    Value = "Option1",
                }
                new ListValueOption
                {
                    Name = "Option2",
                    Value = "Option2",
                }
            },
        })
    )
);
```

Reply:
Fixed.